fluentd/run.sh

```diff
@@ -220,10 +219,13 @@ FILE_BUFFER_PATH=/var/lib/fluentd
 mkdir -p $FILE_BUFFER_PATH
 
 # Get the available disk size.
-DF_LIMIT=$(df -B1 $FILE_BUFFER_PATH | grep -v Filesystem | awk '{print $2}')
+DF_LIMIT=$(df -B1 $FILE_BUFFER_PATH --output=avail | tail -1)
 DF_LIMIT=${DF_LIMIT:-0}
-if [ "${MUX_FILE_BUFFER_STORAGE_TYPE:-}" = "hostmount" ]; then
-    # Use 1/4 of the disk space for hostmount.
+# Check how many mount points share the same filesystem (in case hostmount)
+MCNT=$( mount | grep $( mount | grep "$FILE_BUFFER_PATH" | awk '{print $1}' ) | wc -l )
```

Comment:
Can you give an example of what the output looks like inside the fluentd pod?

Reply:
Sorry, @richm, this is not what you asked for, but these mount points share the hostmount /dev/xvda2.
```
mount: /dev/xvda2 is already mounted or /var/lib/fluentd busy
       /dev/xvda2 is already mounted on /etc/docker
       /dev/xvda2 is already mounted on /var/log
       /dev/xvda2 is already mounted on /usr/share/zoneinfo/UTC
       /dev/xvda2 is already mounted on /dev/termination-log
       /dev/xvda2 is already mounted on /etc/docker-hostname
       /dev/xvda2 is already mounted on /etc/hosts
       /dev/xvda2 is already mounted on /etc/hostname
       /dev/xvda2 is already mounted on /etc/resolv.conf
       /dev/xvda2 is already mounted on /run/secrets
       /dev/xvda2 is already mounted on /var/lib/fluentd
       /dev/xvda2 is already mounted on /etc/sysconfig/docker
       /dev/xvda2 is already mounted on /var/lib/docker
       /dev/xvda2 is already mounted on /var/lib/docker/containers
       /dev/xvda2 is already mounted on /etc/origin/node
       /dev/xvda2 is already mounted on /etc/fluent/configs.d/user

```

Reply:
This seems fragile - e.g. the first line is going to give `mount` instead of `/dev/xvda2`.  Basically you want to know
- what is the device that `$FILE_BUFFER_PATH` is mounted on
- what other paths have mounted this device

I'm not sure, but maybe there is a better way than grepping the output of mount.

Reply:
> This seems fragile - e.g. the first line is going to give `mount` instead of `/dev/xvda2`. Basically you want to know
> 
>     * what is the device that `$FILE_BUFFER_PATH` is mounted on
>     * what other paths have mounted this device
> 
> I'm not sure, but maybe there is a better way than grepping the output of mount.

I totally agree with you.  I also searched it if there's an api to return the storage type or any useful info to get the type, but no luck and ended up this plan B...  Ok, I'll continue searching it.